docs/shared/recipes/plugins/create-package.md

```diff
@@ -0,0 +1,75 @@
+  if (!name) {
+    throw new Error('Please provide a name for the workspace');
+  }
+
+  console.log(`Creating the workspace: ${name}`);
+
+  // This assumes "my-plugin" and "create-my-plugin" are at the same version
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const presetVersion = require('../package.json').version;
+
+  // TODO: update below to customize the workspace
+  const { directory } = await createWorkspace(`my-plugin@${presetVersion}`, {
+    name,
+    nxCloud: false,
+    packageManager: 'npm',
+  });
+
+  console.log(`Successfully created the workspace: ${directory}.`);
+}
+
+main();
+```
+
+By utilizing your custom preset to handle the generation, we can provide equal functionality between `create-nx-workspace` and your new create-package.
+
+This is a plain node script at this point, and you can use any dependencies you wish to handle things like prompting or argument parsing. Keeping dependencies small and splitting out the command line tool from the Nx plugin is recommended, and will help keep your CLI feel fast and snappy.
+
+## Testing your create package
+
+Because your create-my-plugin package will install your plugin package at runtime, they must be published in order to run them and see the results. In order to test your packages without making them publicly available, a `local-registry` target should be present on project in your workspace's root.
+
+This target will start a local registry server using verdaccio, and configure your machine to point to it. This means that you can safely publish, without hitting npm, and test as if you were an end user of your package. The flow should look something like this:
+
+1. Run `nx local-registry` and wait to see the text that is similar to "http address - http://localhost:${port}/" at this point, you should be able to visit that url and verify that the server is running.
+2. In a second terminal, publish your packages. The exact command here will change depending on your setup, but it could look something like `nx run-many -t publish`. You may wish to verify that you are not publishing to the actual npm registry, by running `npm config get registry` first.
+3. In the same terminal you used to publish, run `npx create-my-plugin test-workspace`. This should create a new workspace in the folder "test-workspace".
+4. After validating the behaviour, use ctrl+c or cmd+c to kill the process that is hosting the local registry server in your first terminal. The target should clean up after itself, and `npm config get registry` should point to the real NPM servers again.
+
+## Publishing your create package
+
+Your plugin and create package will both need to be published to NPM to be useable. When the packages were generated they should have included some publishing scripts, but any form should work as expected. If you followed the steps [above](#testing-your-create-package), then you should already be able to publish.
```

Comment:
"any form should work as expected" -> "you can modify the publish task to satisfy your own requirements"